### ISSUE TITLE
Taxonomy items rendered in alphabetical order regardless of hierarchy when autocomplete is selected 

### ISSUE THREAD

**Issue by luciofassio** [OrchardCMS/Orchard#8131]
I'm using Orchard/Laser/1.10.x.
Languages set in Orchard are italian and french, italian is the default one.
I created a hierarchical taxonomy with the import function (in both italian and french) and added a taxonomy field to Page content item.
When autocomplete is set in the taxonomy field the item list is ordered alphabetically regardless of the hierarchy; when autocomplete is off the item list is rendered correctly.
It happens in all the languages.
 
![chrome_2018-10-19_15-50-34](https://user-images.githubusercontent.com/12617219/47282744-b3c19900-d5e0-11e8-8d6b-f3778a8eb288.png)
![chrome_2018-10-19_16-44-19](https://user-images.githubusercontent.com/12617219/47282749-b7edb680-d5e0-11e8-99f5-0cf7d1449828.png)
![chrome_2018-10-22_10-00-11](https://user-images.githubusercontent.com/12617219/47282947-6134ac80-d5e1-11e8-910f-4b1828103991.png)

**Comment by HermesSbicego-Laser** [comment]
Hi @luciofassio, from what I see you're using a branch of our fork, I imagine because you need O1.10 working with a full integrated localization feature. Our branch is an Hybrid: it's 1.10.x + some feature of dev (cloning, concurrency, etc...). 
By the way, from what I see the `dev `branch also is affected by this, see https://sourcegraph.com/github.com/OrchardCMS/Orchard@dev/-/blob/src/Orchard.Web/Modules/Orchard.Taxonomies/Controllers/TagsController.cs?utm_source=share#L52:1.
What I don't know is if that beahaviour (Alphabetical order) is **by design** or **there is an issue** there so we should order by term hierarchy/weight/label...
@sebastienros ?

**Comment by sebastienros** [comment]
Maybe an option in the field? Because I don't think there is a unique solution to this.

**Comment by HermesSbicego-Laser** [comment]
I agree that we need an option,here...More, the autocomplete acts as filter so we need something like https://stackoverflow.com/questions/11432024/jquery-text-input-autocompletion-for-a-tree-of-values
@luciofassio have you any jquery plugin in mind to do that?

**Comment by luciofassio** [comment]
jQueryUI Autocomplete would work well but also what is implemented now does, the issue i see is that when i start typing the item list which is rendered is not properly ordered, a sort option in the taxonomy field would work well i guess
